### DISCUSSION TITLE
nvim-cmp unmap <CR> or set preselect to none?

### DISCUSSION THREAD

**Discussion by gavrh** [NvChad/NvChad#2760] [Q&A] [answered]
### How and in what files can I unmap < CR > from autocomplete or set preselect to none? IOW I want to go to new line without having to do ctrl-e. Current setup is almost default nvchad starter besides theme lsp and relative number config, can anyone help?

I'm new to nvim and am not experienced in lua, I plan to dive deeper into this on my own but I'm stumped on this and can't find much online about it.

### example
![](https://github.com/NvChad/NvChad/assets/96849112/920f2d63-0e15-49a9-b10f-941ac4e0e4a5)

**Comment by siduck** [comment]
@grhx look at the last example of telescope there https://nvchad.com/docs/config/plugins/ 
You have to do the same with cmp i.e import our default config which is at `nvchad.configs.cmp path ( lua module ) 

and then delete <cr> value in the mapping table,  you do that by setting it as `nil` in lua

for example if `person` is a table and then  `person.name = nil`  the value with name key here will be deleted

**Reply by gavrh** [reply]
I added the nvchad config to my project and tried my best to replicate the telescope example with cmp, i'm not sure really if it should be cmp.options.mapping["< CR >"] = ... or something else but I did just create the local variable of nvchad.config.cmp and just returned it so it loaded and removed the "< CR >" mapping from the cmp.lua file and it did the job, if you could let me know about how to actually do it in the init file that would be great but it's working!

**Reply by gavrh** [reply]
nvim/lua/plugins/init.lua
```lua
-- add this vvvv
{
  "hrsh7th/nvim-cmp",
  opts = function()
    local cmp = require "nvchad.configs.cmp"
    return cmp
  end,
}

```

nvim/lua/nvchad/configs/cmp.lua
```lua
mapping = {
  ["<C-p>"] = cmp.mapping.select_prev_item(),
  ["<C-n>"] = cmp.mapping.select_next_item(),
  ["<C-d>"] = cmp.mapping.scroll_docs(-4),
  ["<C-f>"] = cmp.mapping.scroll_docs(4),
  ["<C-Space>"] = cmp.mapping.complete(),
  ["<C-e>"] = cmp.mapping.close(),
-- deleted this vvvv
  ["<CR>"] = cmp.mapping.confirm {
    behavior = cmp.ConfirmBehavior.Insert,
    select = true,
  },
```
this is likely a suboptimal way of doing this but i'm new to nvim and lua and this works so... im using it.

**Reply by siduck** [reply]
please dont edit the nvchad dir directly :/

**Reply by siduck** [reply, marked as answer]
```lua
{
  "hrsh7th/nvim-cmp",
  opts = function()
    local config= require "nvchad.configs.cmp"
    config.mapping["<CR>"] = nil
    return config
  end,
  ```

**Reply by gavrh** [reply]
lol I was guessing that wasnt a good idea, thanks

**Reply by siduck** [reply]
@grhx  :grimacing: 
```lua
{
  "hrsh7th/nvim-cmp",
  config = function(_,opts)
    opts.mapping["<CR>"] = nil
    require("cmp").setup(opts)
  end,
 }
```

**Reply by T3sT3ro** [reply]
Is there some way to define this config in `configs/cmp.lua`??? I am running in some complications with that... (unrelated to this specific mapping from the question, I just want my configs out of the plugins/init.lua)

**Reply by siduck** [reply]
@T3sT3ro  just make your configs/cmp.lua return a function that returns a table or return a table if that file has 0 calls to cmp module ( cuz this will break the lazyloading, thats why need to wrap in a func )
